src/main/java/org/broadinstitute/dsm/model/gbf/GBFOrderGateKeeper.java

```diff
@@ -0,0 +1,49 @@
+package org.broadinstitute.dsm.model.gbf;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+/**
+ * Uses a {@link GBFOrderTransmitter} and {@link GBFOrderFinder}
+ * to find and transmit orders to GBF for a given
+ * DDP instance
+ */
+public class GBFOrderGateKeeper {
+
+    private final GBFOrderFinder orderFinder;
+    private final GBFOrderTransmitter transmitter;
+    private final String ddpInstanceName;
+    private static final String UPDATE_TRANSMISSION_DATE = "update ddp_kit_request req set req.transmitted_at = ? where external_order = ?";
+
+    public GBFOrderGateKeeper(GBFOrderFinder orderFinder, GBFOrderTransmitter transmitter, String ddpInstanceName) {
+        this.orderFinder = orderFinder;
+        this.transmitter = transmitter;
+        this.ddpInstanceName = ddpInstanceName;
+    }
+
+
+    public void sendPendingOrders(Connection conn) {
+        for (SimpleKitOrder simpleKitOrder : orderFinder.findKitsToOrder(ddpInstanceName, conn)) {
+            Response orderResponse = transmitter.orderKit(simpleKitOrder.getRecipientAddress(), simpleKitOrder.getExternalKitName(), simpleKitOrder.getExternalKitOrderNumber(), simpleKitOrder.getParticipantGuid());
+
+            if (orderResponse.isSuccess()) {
+                try (PreparedStatement stmt = conn.prepareStatement(UPDATE_TRANSMISSION_DATE)) {
+                    //stmt.setTimestamp(1, Instant.now());
```

Comment:
I don't think that should be commented out

Reply:
I agree